internal/security/secretstoreclient/vault_test.go

```diff
@@ -49,13 +47,54 @@ func TestHealthCheck(t *testing.T) {
 
 	host := strings.Replace(ts.URL, "https://", "", -1)
 	vc := NewSecretStoreClient(mockLogger, NewRequestor(mockLogger).Insecure(), "https", host)
-	code, _ := vc.HealthCheck()
+	code, err := vc.HealthCheck()
 
+	assert.NoError(t, err)
 	if code != http.StatusOK {
 		t.Errorf("incorrect vault health check status.")
 	}
 }
 
+// TestHealthCheckUninit tests that a no error object is returned
+// for uninitialized vault (with 5xx error)
+func TestHealthCheckUninit(t *testing.T) {
```

Comment:
Add additional test case to ensure that HealthCheck doesn't return Error object when the status code is in the 5xx range.